docs/Deprecating-Disabling-and-Removing-Casks.md

```diff
@@ -0,0 +1,110 @@
+# Deprecating, Disabling and Removing Casks
+
+There are many reasons why casks may be deprecated, disabled or removed. This document explains the differences between each method as well as explaining when one method should be used over another.
+
+## Overview
+
+These general rules of thumb can be followed:
+
+- `deprecate!` should be used for casks that _should_ no longer be used.
+- `disable!` should be used for casks that _cannot_ be used.
+- Casks that are no longer acceptable in `homebrew/cask` or have been disabled for over a year _should_ be removed.
+
+## Deprecation
+
+If a user attempts to install a deprecated cask, they will be shown a warning message but the install will proceed.
+
+A cask should be deprecated to indicate to users that the cask should not be used and will be disabled in the future. Deprecated casks should continue to be maintained by the Homebrew maintainers if they continue to be installable. If this is not possible, they should be immediately disabled.
+
+The most common reasons for deprecation are when the upstream project is deprecated, unmaintained, or archived.
+
+Casks should only be be deprecated if at least one of the following are true:
+
+- the cask does not install on any of our supported macOS versions
+- the cask has outstanding CVEs
```

Comment:
Since we just install casks instead of having to go through trying to build them ourselves, this is acceptable on a second read. I'm not crazy about the unknown with CVE's, but people are supposed to know what they are installing so 🤷‍♂️

Reply:
> I think we should consider including casks with "unpopular" casks here, irrespective of software support or vulnerabilities.

If they have literally zero installs: sure. If they have more than that: why remove them if they aren't causing issues? Less software in homebrew-cask makes it less useful to users.